### ISSUE TITLE
Support token suppression, forced tokens (besides eos and bos), and decoder prompting for flax generation

### ISSUE THREAD

**Issue by andyehrenberg** [huggingface/transformers#20539]
### Feature request

Add logits processors for token suppression and forced tokens at specific indices.
Enable prompting the decoder of encoder-decoder models with decoder_input_ids.

### Motivation

Currently, the flax generation utilities do not support token suppression, forcing specific tokens to be decoded at specific response indices, nor prompting the decoder (helpful for models like Whisper that support decoder prompts - Flax Whisper is implemented in #20479). Adding these would move the flax utilities closer to feature parity with the pytorch generation utilities. Adding these features would fully unlock a flax implementation of Whisper inference.

### Your contribution

I already have these features implemented in a branch of my fork - happy to open a PR!

**Comment by andyehrenberg** [comment]
@sanchit-gandhi @patil-suraj @gante

**Comment by sanchit-gandhi** [comment]
Did we decide to implement these features in the Flax Whisper PR in the end? cc @ArthurZucker

**Comment by andyehrenberg** [comment]
@sanchit-gandhi @ArthurZucker I just added these back into the Flax Whisper PR

**Comment by sanchit-gandhi** [comment]
Cool! Closing this issue in favour of the PR https://github.com/huggingface/transformers/pull/20479
